src/overview/onboarding/components/checklist-container.tsx

```diff
@@ -39,102 +38,90 @@ export type Props = StateProps & DispatchProps & OwnProps
     }
 
-    handlePowerSearchStage = async () => {
+    private handlePowerSearchStage = async () => {
         if (this.props.powerSearchStage === 'DONE') {
             return
         }
 
-        this.processEvent({
-            type: EVENT_NAMES.START_POWERSEARCH_ONBOARDING,
-        })
-
         /*
         If there are no results in Overview, take user to the Wiki
         Else, directly start the onboarding tooltip process.
         */
         if (this.props.noResults) {
-            await this._setOnboardingKey('step2', 'redirected')
-            await this._openDemoPage()
+            await utils.setOnboardingStage('powerSearch', STAGES.redirected)
+            await utils.openDemoPage()
         } else {
-            await this._setOnboardingKey('step2', 'overview-tooltips')
+            await utils.setOnboardingStage(
+                'powerSearch',
+                STAGES.powerSearch.overviewTooltips,
+            )
             this.props.initOnboardingTooltips()
         }
+
+        this.processEvent({
+            type: EVENT_NAMES.START_POWERSEARCH_ONBOARDING,
+        })
     }
 
-    handleTaggingStage = async () => {
+    private handleTaggingStage = async () => {
         if (this.props.taggingStage === 'DONE') {
             return
         }
-        // TODO: Add analytics
-        await this._setOnboardingKey('step3', 'redirected')
-        await this._openDemoPage()
+
+        await utils.setOnboardingStage('tagging', STAGES.redirected)
+        await utils.openDemoPage()
+
+        this.processEvent({
+            type: EVENT_NAMES.START_TAGGING_ONBOARDING,
+        })
     }
 
-    handleBackupStage = async () => {
+    private handleBackupStage = async () => {
         if (this.props.backupStage === 'DONE') {
             return
         }
 
-        // TODO: Add analytics
-        await browser.tabs.create({
-            url: browser.runtime.getURL(constants.BACKUP_URL),
-            active: true,
-        })
+        this.openOptionsTab('backup')
         await this.props.setBackupStageDone()
+
+        this.processEvent({
+            type: EVENT_NAMES.FINISH_BACKUP_ONBOARDING,
+        })
     }
 
     render() {
-        const {
-            showOnboardingBox,
-            annotationStage,
-            powerSearchStage,
-            taggingStage,
-            backupStage,
-        } = this.props
-
-        if (!showOnboardingBox) {
+        if (!this.props.showOnboardingBox) {
             return null
         }
 
         return (
             <Checklist
                 isRightBox={this.props.isRightBox}
                 congratsMessage={this.props.congratsMessage}
-                isAnnotationChecked={annotationStage === 'DONE'}
-                isPowerSearchChecked={powerSearchStage === 'DONE'}
-                isTaggingChecked={taggingStage === 'DONE'}
-                isBackupChecked={backupStage === 'DONE'}
+                isAnnotationChecked={this.props.annotationStage === 'DONE'}
```

Comment:
Rather than checking these here, wouldn't it make more sense to do these in the selectors? You could export more selectors like `isTaggingChecked` which does this equality check for you

Reply:
Thanks. Added extra selectors for these! :+1: